persistence-layer/memory-storage-provider/user-memory-provider.js

```diff
@@ -0,0 +1,65 @@
+const UserProvider = require('../user-provider'),
+    crypto = require('crypto')
+
+function ensureId(model) {
+    if (!model || model.id) return model
+    //generate random 20-bytes identifier
+    model.id = crypto.randomBytes(20).toString('hex')
```

Comment:
That's a source of potential conflict. With extra-small probability, but still... 
See how Mongodb generates unique ids: https://docs.mongodb.com/manual/reference/method/ObjectId/